server.go

```diff
@@ -351,6 +351,13 @@ func (s *Server) attachClient(cl *Client, listener string) error {
 		return fmt.Errorf("ack connection packet: %w", err)
 	}
 
+	// Publish any retained messages for subscriptions which already existed on client takeover.
+	if sessionPresent {
+		for _, sub := range cl.State.Subscriptions.GetAll() {
```

Comment:
As the new client cannot subscribe until the connack has been issued, GetAll can only contain inherited subscriptions.